.github/workflows/lint.yml

```diff
@@ -0,0 +1,27 @@
+name: Lint
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  golangci-lint:
+    name: Run on Ubuntu
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone the code
+        uses: actions/checkout@v4
+
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '~1.22'
```

Comment:
Refer to https://github.com/etcd-io/etcd-operator/blob/f975f942f6e482ca3257bc906087be98cf06c0d1/.github/workflows/test-e2e.yml#L12-L18